src/Init/Lean/Parser/Syntax.lean

```diff
@@ -32,10 +32,10 @@ namespace Syntax
 @[builtinSyntaxParser] def ident     := parser! nonReservedSymbol "ident"
 @[builtinSyntaxParser] def try       := parser! nonReservedSymbol "try " >> syntaxParser
 @[builtinSyntaxParser] def lookahead := parser! nonReservedSymbol "lookahead " >> syntaxParser
-@[builtinSyntaxParser] def optional  := parser! nonReservedSymbol "optional " >> syntaxParser
 @[builtinSyntaxParser] def sepBy     := parser! nonReservedSymbol "sepBy " >> syntaxParser >> syntaxParser
 @[builtinSyntaxParser] def sepBy1    := parser! nonReservedSymbol "sepBy1 " >> syntaxParser >> syntaxParser
 
+@[builtinSyntaxParser] def optional  := tparser! pushLeading >> symbolAux "?" none
```

Comment:
@leodemoura I realized only after the fact that `p?` parses as a single identifier D: ... that is pretty unfortunate. I would still prefer `(p)?` over `(optional p)`, I think.

Reply:
I got disappointed when I tried it too. It is the price of being able to write `x.find?` :(
BTW, I will change the precedence of `try`, `sepBy` nested parsers.
Right now, `try p1 p2` parsers as `try (p1 p2)`.